src/crud_ops.rs

```diff
@@ -44,6 +44,88 @@ impl RawResponse for Response {
     }
 }
 
+// ================================================================================================
+// Boiler plate code for FFI structs
+// Any non-copy fields of ListEntry must be properly destroyed on destroy_list_entries
+#[repr(C)]
+pub struct BulkFailedEntry {
+    path: *const c_char,
+    error_message: *const c_char
+}
+unsafe impl Send for BulkFailedEntry {}
+
+// Only stores paths of entries that the bulk operation failed on
+#[repr(C)]
+pub struct BulkResponse {
+    result: CResult,
+    failed_entries: *const BulkFailedEntry,
+    failed_count: u64,
+    error_message: *mut c_char,
+    context: *const Context
+}
+
+unsafe impl Send for BulkResponse {}
+
+impl RawResponse for BulkResponse {
+    type Payload = Vec<(Path, String)>;
```

Comment:
As discussed elsewhere we will fix the ergonomics of this later, but one thing we can do now is defer converting the error to string to `set_payload`. That way a Rust user of `bulk_delete` still has an `Error` to work with.

```suggestion
    type Payload = Vec<(Path, crate::Error)>;
```